.github/workflows/e2e-tests.yml

```diff
@@ -49,10 +55,15 @@ jobs:
         ./tests/e2e/bin/env.sh start
 
     - name: Run tests
+      if: ${{ matrix.gutenberg == 'no' }}
       run: yarn test-e2e
 
+    - name: Run tests (excluding updater)
+      if: ${{ matrix.gutenberg == 'latest' }}
+      run: yarn test-e2e --testPathIgnorePatterns=updater
+
     - name: Upload Allure artifacts
-      if: ${{ always() }}
+      if: ${{ matrix.gutenberg == 'no' }}
```

Comment:
This might be a behavior change.

If I understand [the docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions) correctly, the `if: always()` would make this step run even if the tests failed. Changing it to `if: matrix.gutenberg == 'no'` will only run it if the tests passed. If we want to keep the previous behavior, we'd have to do like `if: matrix.gutenberg == 'no' && always()`

Reply:
Hmm, I thought that this is the right change: 

Before:

- We ran this job `always` for the without-GB workflow.
- We didn't run it for the with-GB workflow.

Now that we've merged workflows, I've changed the criterion to only run for the without-GB workflow

FWIW, the docs you linked say

> always
>
> Always returns `true`, even when canceled. [...]

That would mean that `matrix.gutenberg == 'no' && always()` is equivalent to `matrix.gutenberg == 'no'` anyway, no?

Reply:
This is the relevant bit:

> You can use the following status check functions as expressions in `if` conditionals. If your `if` expression does not contain any of the status functions it will automatically result with `success()`.

Just `matrix.gutenberg == 'no'` is implicitly equivalent to `matrix.gutenberg == 'no' && success()`, since none of the status functions were included. Adding the `&& always()` avoids the implicit `&& success()`.

Reply:
I did a test at https://github.com/anomiex/testing/pull/1/checks?check_run_id=1539150166. The "If with no status function" step there had `if: ${{ 'a' == 'a' }}` and you can see it skipped running that step. The "If with status function" step had `if: ${{ 'a' == 'a' && always() }}`, and you can see it still ran the step even though the first step in the job failed.

Reply:
Updated accordingly :+1: